File: BackEnd/GPON_Equipment_MicroService/src/main/java/com/example/gpon_equipment_microservice/Service/Board/BoardService.java
package com.example.gpon_equipment_microservice.Service.Board;

import com.example.gpon_equipment_microservice.Entity.Board.Board;
import com.example.gpon_equipment_microservice.Entity.Status;
import com.example.gpon_equipment_microservice.Repository.BoardRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class BoardService implements IBoardService {
    
    private final BoardRepository boardRepository;

    @Override
    public Board save(Board board) {
        return boardRepository.save(board);
    }

    @Override
    public List<Board> findAll() {
        return boardRepository.findAll();
    }

    @Override
    public Optional<Board> findById(Long id) {
        return boardRepository.findById(id);
    }

    @Override
    public List<Board> findByStatus(Status status) {
        return boardRepository.findByStatus(status);
    }

    @Override
    public List<Board> findByOltId(Long oltId) {
        return boardRepository.findByOltId(oltId);
    }
    
    @Override
    public Board findByOltIdAndSlotNumber(Long oltId, int slotNumber) {
        return boardRepository.findByOlt_IdAndSlotNumber(oltId, slotNumber);
    }
    
    @Override
    public Board findByAddressId(Long addressId) {
        return boardRepository.findByAddressId(addressId);
    }

    @Override
    public void deleteById(Long id) {
        boardRepository.deleteById(id);
    }
}

File: FrontEnd/src/Services/Equipment/Board/board-service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environment/environment';
import { Board } from '../../../Core/Models/Equipment/Board';
import { Status } from '../../../Core/Models/Equipment/Status';

@Injectable({
  providedIn: 'root'
})
export class BoardService {
  private apiUrl = `${environment.BASE_URL}/equipment/api/board`;

  constructor(private http: HttpClient) { }

  // Get all boards
  findAll(): Observable<Board[]> {
    return this.http.get<Board[]>(`${this.apiUrl}/all`);
  }

  // Get board by ID
  findById(id: number): Observable<Board> {
    return this.http.get<Board>(`${this.apiUrl}/${id}`);
  }

  // Get boards by status
  findByStatus(status: Status): Observable<Board[]> {
    return this.http.get<Board[]>(`${this.apiUrl}/status/${status}`);
  }

  // Get boards by OLT ID
  findByOltId(oltId: number): Observable<Board[]> {
    return this.http.get<Board[]>(`${this.apiUrl}/olt/${oltId}`);
  }

  // Get board by OLT ID and slot number
  findByOltIdAndSlotNumber(oltId: number, slotNumber: number): Observable<Board> {
    return this.http.get<Board>(`${this.apiUrl}/olt/${oltId}/slot/${slotNumber}`);
  }

  // Get board by address ID
  findByAddressId(addressId: number): Observable<Board> {
    return this.http.get<Board>(`${this.apiUrl}/address/${addressId}`);
  }

  // Save new board
  save(board: Board): Observable<Board> {
    return this.http.post<Board>(`${this.apiUrl}/save`, board);
  }

  // Update existing board
  update(id: number, board: Board): Observable<Board> {
    return this.http.put<Board>(`${this.apiUrl}/update/${id}`, board);
  }

  // Delete board
  delete(id: number): Observable<any> {
    return this.http.delete(`${this.apiUrl}/delete/${id}`);
  }
}
